pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableFactory.java

```diff
@@ -32,7 +34,9 @@ public static DataTable getDataTable(ByteBuffer byteBuffer)
     int version = byteBuffer.getInt();
     switch (version) {
       case 2:
-        return new DataTableImplV2(byteBuffer);
+        return convertDataTableImplV2ToV3(new DataTableImplV2(byteBuffer));
+      case 3:
```

Comment:
can we use the VERSION_ constants defined

Reply:
done